pkg/controller/node/nodecontroller.go

```diff
@@ -371,6 +402,42 @@ func (nc *NodeController) Run() {
 			}
 		}, nc.nodeMonitorPeriod, wait.NeverStop)
 
+		go wait.Until(func() {
+			nc.evictorLock.Lock()
+			defer nc.evictorLock.Unlock()
+			for k := range nc.zoneIntolerablePodEvictor {
+				func(k string) {
+					nc.zoneIntolerablePodEvictor[k].Try(func(value TimedValue) (bool, time.Duration) {
+						message, _ := value.UID.(evictionMessage)
+						podName := message.podName
+						podNamespace := message.podNamespace
+						nodeUID := message.nodeUID
+
+						pod, err := nc.kubeClient.Core().Pods(podNamespace).Get(podName)
```

Comment:
You probably want to get this from `podInformer` instead of getting it from etcd. Double deletions are not a big deal.